pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -797,9 +798,21 @@ public void unloadNamespaceBundlesGracefully() {
                     return this.loadOrCreatePersistentTopic(topicName, createIfMissing);
                 });
             } else {
-                return topics.computeIfAbsent(topic, (topicName) -> {
-                    if (createIfMissing) {
-                        return createNonPersistentTopic(topicName);
+                return topics.computeIfAbsent(topic, (name) -> {
+                    final TopicName topicName = TopicName.get(name);
+                    if (topicName.isPartitioned()) {
+                        try {
+                            final PartitionedTopicMetadata metadata = pulsar().getAdminClient().topics()
+                                    .getPartitionedTopicMetadata(topicName.getPartitionedTopicName());
```

Comment:
Is there a better way for this check?

Reply:
You can use `fetchPartitionedTopicMetadataAsync` in the BrokerService.java, this one will fetch the metadata from the zookeeper cache.

Reply:
thanks. implemented !

Reply:
Cool, if the PR is ready for review, please ping us. I noticed the status of the PR is draft.

Reply:
Now it is ready (y)